File: src/main/java/ru/ifmo/ted/security/XmlAuthority.java
package ru.ifmo.ted.security;


import org.springframework.security.core.GrantedAuthority;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement(name = "authority")
@XmlAccessorType(XmlAccessType.FIELD)
class XmlAuthority implements GrantedAuthority {

    private String authority;

    @Override
    public String getAuthority() {
        return authority;
    }

    public void setAuthority(String value) {
        this.authority = value;
    }
}

File: src/main/kotlin/ru/ifmo/ted/security/XmlAuthority.kt
package ru.ifmo.ted.security

import jakarta.xml.bind.annotation.XmlAccessType
import jakarta.xml.bind.annotation.XmlAccessorType
import jakarta.xml.bind.annotation.XmlElement
import jakarta.xml.bind.annotation.XmlRootElement
import org.springframework.security.core.GrantedAuthority

@XmlRootElement(name = "authority")
@XmlAccessorType(XmlAccessType.FIELD)
class XmlAuthority(private val authority: String) : GrantedAuthority {
    @XmlElement
    override fun getAuthority() = authority
}